app/services/recurring_contribution/subscriptions/cancel_expired.rb

```diff
@@ -0,0 +1,38 @@
+class RecurringContribution::Subscriptions::CancelExpired
+  attr_reader :subscriptions
+
+  def initialize(subscriptions)
+    @subscriptions = subscriptions
+  end
+
+  def self.process(subscriptions)
+    new(subscriptions).process
+  end
+
+  def process
+    pagarme_subscriptions = Pagarme::API.retrieve_pagarme_subscriptions(subscriptions.keys)
+    canceled_subscriptions = cancelable_subscriptions(pagarme_subscriptions, subscriptions)
+
+    return if canceled_subscriptions.empty?
+    Pagarme::API.cancel_subscriptions(canceled_subscriptions)
+  end
+
+  private
+
+  def cancelable_subscriptions(pagarme_subscriptions, subscriptions)
+    [].tap do |canceled_subscriptions|
+      pagarme_subscriptions.each do |pagarme_subscription|
+        next if pagarme_subscription[:errors].present?
+
+        subscription = subscriptions[pagarme_subscription[:id]]
+
+        canceled_subscriptions << subscription if cancelable?(pagarme_subscription[:charges], subscription)
+      end
+    end
+  end
+
+  def cancelable?(pagarme_charges, subscription)
+    (subscription.bank_billet? && pagarme_charges == subscription.charges) ||
```

Comment:
It's just a suggestion, you can improve it:

```ruby
def cancellable?(charges, subscription)
 return cancellable_bank_billet(charges, subscription) if subscription.bank_billet?
 return cancellable_credit_card?(charges, subscriptions) if subscription.credit_card?
end

def cancellable_bank_billet?(charges, subscriptions)
 charges == subscriptions.charges
end

def cancellable_credit_card?(charges, subscriptions)
 charges == subscriptions.charges - 1
end
```
Improve method names and choose a name for this constant `1`.